node-phantom.js

```diff
@@ -227,6 +227,11 @@ module.exports = {
         var prematureExitHandler = function (code, signal) {
           console.warn('phantom crash: code ' + code)
           server.close()
+
+          // required to signal up the call stack that phantom has crashed
+          // the catching block will signal a error return code allowing
+          // a supervisor process to restart
+          throw("premature phantom crash");
```

Comment:
OK WFM I thought you meant at first that the only way to signal is through a JS exception which I found odd.  But it's just that this will cause an exit of the process with non zero return code. OK so I'm fine with it in any case.